Types.h

```diff
@@ -58,13 +59,15 @@ typedef struct SBotConfig
     std::string Path;
     sc2::Race Race;
 	sc2::Difficulty Difficulty;
+	std::string Args; //Optional arguments
     SBotConfig() {}
-    SBotConfig(BotType InType, std::string InBotName, sc2::Race InBotRace, std::string InBotPath, sc2::Difficulty InDifficulty = sc2::Difficulty::Easy)
+    SBotConfig(BotType InType, std::string InBotName, sc2::Race InBotRace, std::string InBotPath, sc2::Difficulty InDifficulty = sc2::Difficulty::Easy, std::string InArgs="")
```

Comment:
Please use references in order to avoid redundant object creation.

Reply:
I am still new to C++. The fields in the struct would be still real copies if I use references, right?

Reply:
The fields will be a kind of pointers to the real objects.
Without of reference an object will be copied at least twice if I not mistaken before it will make it's way to the functions body.

Reply:
Yeh, I intend to replace a lot of things like that when I go and do a bit of a code cleanup

Reply:
But if they are pointer and I do something like this
```
std::vector<SBotConfig> configs;
while (we still have to create bots)
    {
    std::string botname = getBotName();
    configs.push_back(SBotConfig(...,botname,...));
    }
std::cout << configs.front().BotName << std::endl;
```
Wouldn't the ```BotName``` field be empty, because ```botname``` went out of scope? I don't think so because they ARE real copies, i.e., the function gets the reference (no copy) and the field then gets a real copy, right? Sorry if this is a stupid question :)

Reply:
Well, it is a kind of pointer, not real pointer) You example will work well.
The only trick with reference here is to get rid of redundant copying, all the other operations (e.g. copying the value into a config) are safely can be done. And you will don't see any difference in the function's body (i.e. all the work done as it was a copy of an original object) because the language will hide it from you.